linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -1091,6 +1098,53 @@ private final class ClassDefChecker(classDef: ClassDef,
     }
   }
 
+  private def checkLinkTimeTree(tree: Tree, expectedType: PrimType): Unit = {
```

Comment:
I'm wondering if it makes sense to keep this or if we should change `tryEvalLinkTimeIf` to support returning the relevant context on and just forward to it.

That would probably remove quite some duplication in terms of checking what is allowed and what isn't with minimal to no performance impact because we expect these trees to be tiny.